### PR TITLE
Merge Sgen to Release/2.0.0

### PR DESCRIPTION
**Pull request by shmao**
The PR is to merge the code for Sgen to 2.0.0 branch.

### CODE REVIEW

**Comment by shmao** [comment]
@huanwu can you please review the list of the commits?

**Comment by weshaggard** [comment]
@shmao please make sure you send this to netcoreship for approval. It isn't clear from the changes but does this impact any other libraries beyond the sgen tool? Will they all need to be reshipped or will they be impacted if we do need to service them for other reasons?

**Comment by shmao** [comment]
@weshaggard Thanks for the reminder. I'll send a request to netcoreship. No other library needs to be reshipped. And this shouldn't impact any library if we need to service them.

**Comment by shmao** [comment]
@weshaggard we got approval from the shiproom. Can we remove the `No Merge` label and complete the PR?

File: src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs [hunk 1]
```diff
@@ -173,6 +173,11 @@ internal static Assembly LoadGeneratedAssembly(Type type, string defaultNamespac
                 name.CodeBase = null;
                 name.CultureInfo = CultureInfo.InvariantCulture;
                 string serializerPath = Path.Combine(Path.GetDirectoryName(type.Assembly.Location), serializerName + ".dll");
+                if (!File.Exists(serializerPath))
```
Comment (weshaggard) [inline src/System.Private.Xml/src/System/Xml/Serialization/Compilation.cs]: Isn't this touching code in the Xml library? Don't we need to worry about the impact of such changes on the shipping code outside of this tool?

**Comment by weshaggard** [comment]
I removed the label. But I must say I do worry a little about the next servicing event picking up more changes in other libraries like System.Private.Xml as an example. We may want to consider #ifdef'ing those.

**Comment by shmao** [comment]
Thanks for reviewing the code changes, @weshaggard. Most of the changes in System.Private.Xml are already under `#if XMLSERIALIZERGENERATOR`. The change you pointed out above was in the method named `LoadGeneratedAssembly` which was used by the SGen tool only. The impact of the changes to System.Private.Xml is minimized if there's any.

cc: @zhenlan @huanwu

**Pull request merged**